### ISSUE TITLE
"wmic" does not exist

### ISSUE THREAD

**Issue by Chris122AT** [McAlex777/Datenschutz#2]
Hello, 

if I run your script Datenschutz.bat I get the following error instead of "Parameter OK.":

[German]:
`Der Befehl "wmic" ist entweder falsch geschrieben oder konnte nicht gefunden werden.`

Windows 10 Home: 10.0.19045 (German)

Does this have any impact on the correct function?

Thank you!
Chris

**Comment by McAlex777** [comment]
Hello,
in a first look I would say, I do use "wmic" for getting only date/time for directorys and logfiles.

you can try find out follow command:
`wmic OS Get localdatetime`

Result:
https://i.ibb.co/7br0zLJ/WMIC.png

Ill check against "Windows10 Home 19045" next days, and will give first january-week details or update.

**Comment by Chris122AT** [comment]
Hello,

the issue occurs in version 1.3.2.

These are the results of running following commands as administrator:

`wmic OS Get localdatetime`
`Der Befehl "wmic" ist entweder falsch geschrieben oder konnte nicht gefunden werden.`

`c:\Windows\System32\wbem\WMIC.exe OS Get localdatetime`
```
LocalDateTime              
20221227112808.154000+060
```

I added c:\Windows\System32\wbem\ temporarily to the PATH environment variable, which fixed the issue:
`SET PATH=%PATH%;c:\Windows\System32\wbem\`

The result instead of much warnings:

```
[...]

Windows 10 HOME 22H2 Lang: de-DE Job: Registry

Parameter OK.

[...]
```

So, if you add SET PATH=%PATH%;c:\Windows\System32\wbem\ in your Datenschutz.bat it should work as expected on Windows systems, who don't load WMIC in PATH environment variable by default.

Notice: WMIC is obsolete.

`wmic /?`
`WMIC ist veraltet.`

You may want to replace the command in future releases.

**Comment by McAlex777** [comment]
Hello,
My reasons for wmic:
Date/time results different outputs in different windows-languages, wmic is standardized.
Wmic-commands are significantly faster then powershell-commands.
Therefore I also prefer batch over powershell.

I know wmic is obsolete, but as long its installed by default - and there are no issues ill use it.
I didnt saw problems with %PATH% variable in all my tests. 

To fix this problem, you have todo same PATH-Workarround in "fwi.bat", and call "Firewall_Install.bat" again.
(if you are using this script)

In next version in january/february ill fix this finaly.
e.g. setting this %PATH% automatically when wmic not found, "and" exists in path "c:\Windows\System32\wbem\"
e.g. Using Powershell when wmic isnt in "c:\Windows\System32\wbem\" also

So, you can use this script in this way, and in next version problem willbe fixed automatically.

Many Thanks for feedback
Alex

**Comment by Chris122AT** [comment]
Hi Alex,
that makes sense using a standardized on board tool regardless of whether it is out of date or not.

Currently I don't use your fwi.bat because my firewall rules were set with [WindowsSpyBlocker](https://github.com/crazy-max/WindowsSpyBlocker). I didn't compare your bundle of firewall rules with the IP block lists from WindowsSpyBlocker, because it seems the way of working is different.

Do you know WindowsSpyBlocker? Does it make sense using your fwi script combined with WindowsSpyBlocker?

All the best
Chris

**Comment by McAlex777** [comment]
Hello,
I don't know "WindowsSpyBlocker" and don't know how it works.

"Privacy.bat" deactivates all telemetry with Microsoft/Windows board means as far as this is reasonably possible. Primarily the Microsoft documentation and SiSyPHuS are used. 

"Datenschutz.bat" deactivates the telemetry services so that they no longer run. "Datenschutz.bat" additionally reduces the amount of telemetry data via policy configuration. The firewall rules set by "Datenschutz.bat" serve as a third additional security layer.

"fwi.bat" has 2 tasks:
1.) Load a self-defined Windows firewall ruleset.
2.) Reset the Windows telemetry settings set by "Datenschutz.bat" every time it is called.

to 1.)
Firewall rules are only loaded as soon as the user himself has called up "fwi export". I.e. as long as "fwi export" has not been called manually, no firewall rules are changed. Fwi is based 100% on the existing Windows firewall and does not use any other mechanisms. Fwi does not change any firewall rules (except those of "Datenschutz.bat"), but primarily serves to export/import its own rules. "fwi help" shows more details.

to 2.)
This ensures that "possible" system changes are reset after function updates.

Windows is increasingly becoming a cloud product. That is why, in my view, more than a reduction of telemetry is not feasible with the means that Microsoft itself proposes. 

Best regards
Alex

**Comment by McAlex777** [comment]
"wmic" does not exist > Fixed in v1.3.3.
